sympy/calculus/util.py

```diff
@@ -1631,3 +1632,210 @@ def union(self, other):
 
 # setting an alias for AccumulationBounds
 AccumBounds = AccumulationBounds
+
+
+def argmax_real(f, symbol=None, domain=S.Reals):
```

Comment:
`argmax` sounds better. Sets like, `N` aren't perfectly real. For sets which are not subsets of real a `NotImplementedError` can be raised.